durabletask/task.py

```diff
@@ -313,6 +296,92 @@ def get_completed_tasks(self) -> int:
         return self._completed_tasks
 
 
+class CompletableTask(Task[T]):
+
+    def __init__(self):
+        super().__init__()
+        self._retryable_parent = None
+
+    def complete(self, result: T):
+        if self._is_complete:
+            raise ValueError('The task has already completed.')
+        self._result = result
+        self._is_complete = True
+        if self._parent is not None:
+            self._parent.on_child_completed(self)
+
+    def fail(self, message: str, details: pb.TaskFailureDetails):
+        if self._is_complete:
+            raise ValueError('The task has already completed.')
+        self._exception = TaskFailedError(message, details)
+        self._is_complete = True
+        if self._parent is not None:
+            self._parent.on_child_completed(self)
+
+
+class RetryableTask(CompletableTask[T]):
+
+    def __init__(self, retry_policy: RetryPolicy, action: pb.OrchestratorAction,
+                 is_sub_orch: bool) -> None:
+        super().__init__()
+        self._action = action
+        self._retry_policy = retry_policy
+        self._retry_count = 0
+        self._first_attempt: datetime
```

Comment:
Let's initialize the `_first_attempt` variable so that we don't have any uninitialized fields.

Reply:
I have renamed it to `_start_time`, as it is basically the time at which it starts and is used to calculate the retry_expiration time.
And, instead of setting it with a setter (which was required), passed it while initializing RetryableTask, so it is initialized with required value.